app/src/main/java/com/asfoundation/wallet/ui/widget/holder/TransactionDetailsHolder.java

```diff
@@ -61,18 +67,28 @@ public TransactionDetailsHolder(View view, OnMoreClickListener listener) {
     String currency = addition.getString(DEFAULT_SYMBOL_ADDITIONAL);
 
     String peer = operation.getFrom();
-    int peerLabel = R.string.label_from;
+    int peerLabel = string.label_from;
     // Check if the from matches the current wallet address, ifo so then we change a label to "To"
     if (peer.toLowerCase()
         .equals(defaultAddress)) {
       peer = operation.getTo();
-      peerLabel = R.string.label_to;
+      peerLabel = string.label_to;
     }
 
     more.setOnClickListener(this);
 
-    fill(operation.getTransactionId(), peerLabel, peer,
-        operation.getFee() + " " + currency.toUpperCase());
+    fill(operation.getTransactionId(), peerLabel, peer, formatFee() + " " + currency.toUpperCase());
+  }
+
+  private String formatFee() {
+    int decimals = BuildConfig.DEBUG ? BuildConfig.ROPSTEN_DEFAULT_TOKEN_DECIMALS
+        : BuildConfig.MAIN_NETWORK_DEFAULT_TOKEN_DECIMALS;
+    long longFee = Long.parseLong(operation.getFee());
+    double doubleFee = longFee / Math.pow(10, decimals);
+
+    return BigDecimal.valueOf(doubleFee)
+        .setScale(DEFAULT_SCALE, RoundingMode.HALF_UP)
+        .toPlainString();
```

Comment:
you can convert the string directly to a BigDecimal and perform the division there, like: 

```
return new BigDecimal(operation.getFee())
        .divide(BigDecimal.valueOf(Math.pow(10.0, decimals)), DEFAULT_SCALE, RoundingMode.HALF_UP)
        .toPlainString();
```

although I don't have a problem with your approach, and if you prefer it, then you can leave it as is

Reply:
Yes you're right. I thought about it but was already using primitives. I prefer your approach though, I'm going to change it.